### ISSUE TITLE
[Enhancement]: Allow importing of chapters from MP3 Title Tags

### ISSUE THREAD

**Issue by ZLoth** [advplyr/audiobookshelf#2816]
### Describe the feature/enhancement

The example book I'm using is B002UZLO80 - The Modern Scholar - The American Legal Experience . In a typical process, I pull and convert the audiobook from Audible and then convert it to MP3 files which are split up by chapter. For this book example, the data from Audible for the chapters list only the chapters, thus I have manually edited the MP3 files to include a Title for each chapter. This is a holdover from when I was utilizing Emby. 

Here is the screen shot from OpenAudible showing that only chapters are being downloaded from Audible being an older title:
![image](https://github.com/advplyr/audiobookshelf/assets/6700159/b5d6931e-9c24-4942-b54c-086780e61850)

Here is a screenshot from MP3Tag of that book. Notice that I have added chapter titles based upon the provided PDF book:
![image](https://github.com/advplyr/audiobookshelf/assets/6700159/578b85e0-70c7-4493-845a-fb96f9951aa3)

The filename is set up to be _track number_ - _book title_ - _chapter title_.mp3 . Because of file naming limitations, the colon is stripped out.

However, when I look at this title in Audiobookshelf 2.8.1:
![image](https://github.com/advplyr/audiobookshelf/assets/6700159/82f2d261-38a6-4197-8bb8-2f081440a1a0)

Somehow, there is a merging of data, and the chapter titles are being duplicated. However, if I do the "Set Chapters From Tracks":

![image](https://github.com/advplyr/audiobookshelf/assets/6700159/29d3fcb9-491b-4be5-b851-0bd1515a7bf8)

Now, I have to go through and edit the titles which is a time-consuming process. It would be nice for Audiobookshelf to pull the MP# title tags and import them into the database. It would be especially handy if I can tag multiple books and update the chapter data.

**Comment by advplyr** [comment]
When the book is first scanned in it should be using the ID3 title tag for the chapter names so long as the mp3 files don't have embedded chapters. Is OpenAudible embedding the full list of chapters in every mp3 file?

**Comment by ZLoth** [comment]
Conducting some tests with a free book - B0C5JFK69Z - Lonely Harts Radio - A Novel... here is the chapter list:
![image](https://github.com/advplyr/audiobookshelf/assets/6700159/85c2a271-c1e1-40b8-ad6b-0b94ec4cdae7)

Screenshot from MP3Tag:
![image](https://github.com/advplyr/audiobookshelf/assets/6700159/f25cce0d-2b0a-47d3-a104-5b2fd8311acb)

Here is the chapter list after doing a scan:
![image](https://github.com/advplyr/audiobookshelf/assets/6700159/568908f3-6f54-4475-ab4b-b5a022f8e9b3)

So it's grabbing both the MP3 tags _and_ the Amazon Metadata. 

But here is the actual metadata.... tags are missing...
![image](https://github.com/advplyr/audiobookshelf/assets/6700159/0ae482bb-8728-46f3-8da9-8a68c179fb0b)

which are these....
![image](https://github.com/advplyr/audiobookshelf/assets/6700159/5f4b976d-395a-4e66-ba30-f4858f94fde8)

**Comment by advplyr** [comment]
Embedded chapters take precedence over the individual track titles and I can see from the green checkmark that the audio files have embedded chapters. Is OpenAudible embedding the full list of chapters in every mp3 file?

You can take a screenshot of the audio file probe

**Comment by ZLoth** [comment]
Yes, OpenAudible is embedding the tags as both part of a single MP3 file...

![image](https://github.com/advplyr/audiobookshelf/assets/6700159/faadaeda-9534-4b7e-aeeb-3b84d8bc1c34)

As well as the split MP3.

**Comment by advplyr** [comment]
If you don't embed chapters in the mp3 files then it will use the ID3 title tag from the files to build the chapter list when first scanning in.

The "Set chapters from tracks" button could be revised to use the ID3 title tags instead of the filename but I have to think about how to update the UI so this is intuitive.

**Comment by ZLoth** [comment]
> The "Set chapters from tracks" button could be revised to use the ID3 title tags instead of the filename but I have to think about how to update the UI so this is intuitive.

I would do this as a set of tabs similar to how you edit a audiobook now. The tabs would be:

- Chapter Editor
- Match from Lookup
- Match from Title Tags
- Match from Filenames

**Comment by nichwall** [comment]
I tried to come up with a better chapter editor page if you'd like to help improve it. I made it into a new discussion so it's easier to combine the chapter editor page discussions. https://github.com/advplyr/audiobookshelf/discussions/3011
